xilem_core2/src/element.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! The types which can be used as elements in a [`View`](crate::View)
+
+/// A type which can be used as the `Element` associated type for a [`View`](crate::View).
+///
+/// It is expected that most libraries using `xilem_core` will have a generic
+/// implementation of this trait for their widget type.
+/// Additionally, this may also be implemented for additional types.
+/// In Xilem (the user interface library)
+///
+/// This does require the reference type to be reborrowable.
+///
+/// In most cases, there will be a corresponding implementation of [`SuperElement<Self>`] for
+/// some other type.
+/// This will be the generic form of this type, which is used for the implementation of [`AnyView`].
+///
+/// [`AnyView`]: crate::AnyView
+///
+// TODO: Rename so that it doesn't conflict with the type parameter names
+pub trait Element {
+    type Mut<'a>;
```

Comment:
Could you comment this associated type - what is it?

Reply:
Added - does that makes sense to you?